src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverter.cs

```diff
@@ -0,0 +1,255 @@
+                    return;
+
+                case ValueType.TimeSpan:
+                    var timeSpan = (TimeSpan)value;
+                    scope.Write(field.Id, timeSpan.ToString("c", CultureInfo.InvariantCulture), ref entryWriter);
+                    return;
+
+                case ValueType.Convertible:
+                    scope.Write(field.Id, ((IConvertible)value).ToString(CultureInfo.InvariantCulture), ref entryWriter);
+                    return;
+
+                case ValueType.Enumerable:
+                    var iterator = (IEnumerable)value;
+                    if (scope.Allocate(field.Id, false) == false)
+                        return;
+                    foreach (var item in iterator)
+                    {
+                        InsertRegularField(field, item, indexContext, out _, ref entryWriter, scope);
+                    }
+                    scope.WriteCollection(field.Id, ref entryWriter);
+                    return;
+
+                case ValueType.DynamicJsonObject:
+                    HandleObject((BlittableJsonReaderObject)value, field, indexContext, out _, ref entryWriter, scope);
+                    return;
+
+                case ValueType.ConvertToJson:
+                    var val = TypeConverter.ToBlittableSupportedType(value);
+                    if (!(val is DynamicJsonValue json))
+                        InsertRegularField(field, val, indexContext, out _, ref entryWriter, scope, nestedArray);
+                    return;
+
+                case ValueType.BlittableJsonObject:
+                    HandleObject((BlittableJsonReaderObject)value, field, indexContext, out _, ref entryWriter, scope);
+                    return; 
+
+                case ValueType.BoostedValue:
+                case ValueType.Stream:
+                case ValueType.DynamicNull:
+                case ValueType.Null:
+                default:
+                    throw new NotImplementedException();
+            }
+
+            shouldSkip = true;
+        }
+
+        void HandleArray(IEnumerable itemsToIndex, IndexField field, JsonOperationContext indexContext, out bool shouldSkip, ref IndexEntryWriter entryWriter, CoraxWriterScope scope, bool nestedArray = false)
+        {
+            shouldSkip = false;
+            if (nestedArray)
+            {
+                return;
+            }
+
+            foreach (var itemToIndex in itemsToIndex)
+            {
+                InsertRegularField(field, itemToIndex, indexContext, out _, ref entryWriter, scope);
+            }
+        }
+
+        //todo maciej Discuss how we gonna handle nestedArrays. Now I skip them.
+        void HandleObject(BlittableJsonReaderObject val, IndexField field, JsonOperationContext indexContext, out bool shouldSkip, ref IndexEntryWriter entryWriter, CoraxWriterScope scope, bool nestedArray = false)
+        {
+            if (val.TryGetMember("$values", out var values) &&
```

Comment:
we need to extract $values to some contants? Would like to avoid doing Ctrl + F to look for magic strings.

Reply:
Found Raven.Client.Constants.Json.Fields.Type.Values